### DISCUSSION TITLE
Are the custom Solid Configs really necessary?

### DISCUSSION THREAD

**Discussion by jeswr** [comunica/comunica#1152] [General]
In most Solid-based libraries we generally pass around the `fetch` function rather than the `Session` (e.g. [solid-client](https://docs.inrupt.com/developer-tools/api/javascript/solid-client/)).

It is also possible to do this with the `KeysHttp.fetch` key in the default Comunica build. That is, the following works:

```ts
const quads = await engine.queryBindings(query, {
      sources: [
        `https://fragments.ap.inrupt.com/qpf?storage=${encodeURIComponent('https://storage.ap.inrupt.com/311cdda9-cb55-465b-861d-5b1e4dc6ea38/')}`,
      ],
      [KeysHttp.fetch.name]: session.fetch
});
```

Is there any reason to have a custom build for Solid given this is the case, if not it might be worth updating the docs to encourage users to migrate to the above usage.

**Comment by rubensworks** [comment]
Long-term, passing sessions makes more sense for optimization opportunities I think.

For example, the engine could cache private documents based on sessions (which is not as easy with custom fetch functions).
Furthermore, the engine could derive some query-relevant information from the session, such as the user's WebID.
This WebID could for instance be useful in link traversal contexts, as a seed URL.

Does that make sense?
